### PR TITLE
Add statistics function for #97

### PR DESCRIPTION
**Pull request by JBlond**

### CODE REVIEW

File: lib/jblond/Diff/Similarity.php [hunk 1]
```diff
@@ -223,4 +223,39 @@ private function ratioReduce(int $sum, array $triple): int
...
+     * Get diff statistics
+     *
+     * @return array
+     */
+    public function getDifference(): array
+    {
+        $return = [
+            'inserted'     => 0,
+            'deleted'      => 0,
+            'notModified'  => 0,
```
Comment (DigiLive) [inline lib/jblond/Diff/Similarity.php:237]: Array index is immediately overwritten before accessing

**Comment by DigiLive** [comment]
We have to think about what "Not Modified" means.
Currently it means all lines which aren't replaced or deleted. (White lines of base version)
Should a deleted/inserted line count as modified or not?

**Comment by JBlond** [comment]
The white lines are not visible in an unfined diff. It might be safe to remove that from the stats.

File: lib/jblond/Diff/Similarity.php (hunk 1 above)
Comment (JBlond) [reply lib/jblond/Diff/Similarity.php:237]: I don't understand what you mean.

**Comment by DigiLive** [comment]
> 
> 
> The white lines are not visible in an unfined diff. It might be safe to remove that from the stats.

The stats should be independent from whats visible.
I was referring to them to clarify what I meant with "not modified".

We should define what "not modified" means.
Line...

1. is (partly) replaced. **n**!
2. is removed from base. y/**n**?
   (Is a completely removed line, a modified line?)
3. is added to new. y/**n**?
   (Doesn't exist in the base version, so can't be modified?)

In the above list, the bold selection is currently set.

**Comment by JBlond** [comment]
I'm not satisfied with the counting.

The example/example.php output has 

```
Between the two versions, there's a 72% match.
Inserted lines: 1
Deleted lines: 1
Lines with replacement: 6
```

But it should be

Inserted lines: 1
Deleted lines: 2
Lines with replacement: 7

**Comment by JBlond** [comment]
```
diff -u a.txt b.txt | diffstat                                                                                                     104ms [15:14]
 b.txt |   19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)
```

**Comment by JBlond** [comment]
my bad, I just made a debug run, and that that it is "equal" in the switch rather than "notModified"

File: lib/jblond/Diff/Similarity.php [hunk 2]
```diff
@@ -223,4 +223,39 @@ private function ratioReduce(int $sum, array $triple): int
...
+     * Get diff statistics
+     *
+     * @return array
+     */
+    public function getDifference(): array
+    {
+        $return = [
+            'inserted' => 0,
+            'deleted'  => 0,
+            'equal'    => 0,
```
Comment (DigiLive) [inline lib/jblond/Diff/Similarity.php:237]: Array index is immediately overwritten at line 257 before accessing

**Review by DigiLive — CHANGES_REQUESTED** [review]

**Review by DigiLive — APPROVED** [review]

**Pull request merged**